diff -up pciutils-3.5.6/lspci.c.freefix pciutils-3.5.6/lspci.c
--- pciutils-3.5.6/lspci.c.freefix	2018-08-21 16:52:35.087706662 +0200
+++ pciutils-3.5.6/lspci.c	2018-08-21 16:52:35.124706469 +0200
@@ -128,6 +128,7 @@ scan_device(struct pci_dev *p)
       fprintf(stderr, "lspci: Unable to read the standard configuration space header of device %04x:%02x:%02x.%d\n",
 	      p->domain, p->bus, p->dev, p->func);
       seen_errors++;
+      free(d);
       return NULL;
     }
   if ((d->config[PCI_HEADER_TYPE] & 0x7f) == PCI_HEADER_TYPE_CARDBUS)
